src/seals/diagnostics/proc_goal.py

```diff
@@ -0,0 +1,62 @@
+"""Gridworld with large random initial state."""
+
+from gym import spaces
+import numpy as np
+
+from seals import base_envs, util
+
+
+class ProcGoalEnv(base_envs.ResettableMDP):
+    """Gridworld with large random initial state.
+
+    In this task, the agent starts at a random position in a large
+    grid, and must navigate to a goal randomly placed in a
+    neighborhood around the agent.  The observation is a 4-dimensional
+    vector containing the (x,y) coordinates of the agent and the goal.
+    The reward at each timestep is the negative Manhattan distance
+    between the two positions.  With a large enough grid, generalizing
+    is necessary to achieve good performance, since most initial
+    states will be unseen.
+    """
+
+    def __init__(self, bounds: int = 100, distance: int = 10):
+        """Constructs environment.
+
+        Args:
+            bounds: the absolute values of the agent's coordinates are
```

Comment:
It's only the agent's *initial* coordinates right? The grid itself is unbounded I think (you do not provide bounds to `grid_transition_fn`)?